### PR TITLE
Workaround for #336

### PR DESCRIPTION
**Pull request by as3boyan**
Workaround for #336

### CODE REVIEW

File: src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxeNamedComponent.java [hunk 1]
```diff
@@ -96,6 +96,15 @@ public ItemPresentation getPresentation() {
...
         final StringBuilder result = new StringBuilder();
+
+        if (AbstractHaxeNamedComponent.this instanceof HaxeLocalVarDeclarationPart) {
+          result.append(AbstractHaxeNamedComponent.this.getName());
+          result.append(":");
+          result.append(HaxeTypeResolver.getFieldOrMethodReturnType(AbstractHaxeNamedComponent.this).toString());
+          return result.toString();
+        }
+
+        //TODO: implement HaxeFieldModel or analogous model for local var declaration part
```
Comment (as3boyan) [inline src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxeNamedComponent.java]: Just a workaround, probably still needs model

**Comment by as3boyan** [comment]
Also return types doesn't seem to be working properly.

**Comment by EBatTiVo** [comment]
I'm unclear.  Do you want to merge the workaround, or do you want to fix this bug properly?  And, if you do want to merge it, please surround the code with comment markers so that folks who come after will know that it's temporary code.

**Comment by as3boyan** [comment]
I don't know how to solve this properly, HaxeFieldModel is based on HaxeVarDeclaration, what we should do for HaxeLocalVarDeclaration?
@soywiz could you please help us with this?

**Comment by EBatTiVo** [comment]
@soywiz - Carlos, can you please review and comment?  Thanks.

**Comment by soywiz** [comment]
Sorry. Didn't noticed this. I'm going to read this now.

**Comment by soywiz** [comment]
https://github.com/TiVo/intellij-haxe/blob/WorkaroundFor336/src/common/com/intellij/plugins/haxe/model/HaxeMemberModel.java#L91

Maybe `HaxeMemberModel.fromPsi` should handle HaxeLocalVarDeclarationPart?

There are regression tests for this?

I would put that workaround, and then would work on a proper fix.

--

Maybe creating a LocalVarModel or ParameterModel or something if not having sense to extend HaxeMemberModel maybe we could add an interface or a superior abstract class. Maybe something lile HaxeTraitModel or HaxeTypedElementModel? Not sure about the name.

That part just uses the .getName() method so maybe am interface like Named would do the work?

https://github.com/TiVo/intellij-haxe/blob/WorkaroundFor336/src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxeNamedComponent.java#L122
```
HaxeTypeResolver.getFieldOrMethodReturnType(AbstractHaxeNamedComponent.this)
```

Maybe this could be extracted from the model, so something like HaxeTypedElementModel would have getName() and getType()? And you would obtain a model of that kind from the psi element.

**Comment by EBatTiVo** [comment]
@as3boyan Are there any unit tests that cover this change?

**Comment by EBatTiVo** [comment]
NOTE: This PR may no longer be neccessary.  PR #456 has supposedly fixed the issue.  Need to verify and close.

**Comment by EBatTiVo** [comment]
@bynuff has fixed the underlying problem with #456.

**Pull request closed without merging**